### PR TITLE
refak(CI): convert the unity fixture to use scenario classes. part 1

### PR DESCRIPTION
**Pull request by rich-bugsnag**
## Goal

This is the first part of a multipart refactor of the way CI runs in the Unity repo.

The end result will be all tests running with scenario classes and all platforms running using mazerunner commands.

## Design

There will be a csharp features folder that contains tests that should run on all platforms.

Each platform will have it's own feature files to test platform specific stuff, similar to how it is now but better organised.

## Changeset

- Added scenario classes and code to find/run these classes

## Testing

The new tests files will run on windows and MacOS for now, later the mobile fixture will be converted to use maze commands and then run the new features.

### CODE REVIEW

File: features/fixtures/maze_runner/Assets/Scripts/Scenarios/Fallback/Breadcrumbs/DisableBreadcrumbs.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+{
+    public override void PrepareConfig(string apiKey, string host)
+    {
+        base.PrepareConfig(apiKey, host);
+        Configuration.EnabledBreadcrumbTypes = new BreadcrumbType[0];
+    }
+
+    public override void Run()
+    {
+        base.Run();
```
Comment (nickdowell) [inline features/fixtures/maze_runner/Assets/Scripts/Scenarios/Fallback/Breadcrumbs/DisableBreadcrumbs.cs:18]: I think you can remove all these calls to `base.Run()`, they just add noise.

File: features/fixtures/maze_runner/Assets/Scripts/Scenarios/Fallback/Breadcrumbs/EnableBreadcrumbs.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using BugsnagUnity;
+using System;
```
Comment (nickdowell) [inline features/fixtures/maze_runner/Assets/Scripts/Scenarios/Fallback/Breadcrumbs/EnableBreadcrumbs.cs:5]: I suspect we don't need all these `using` statements

File: features/fixtures/maze_runner/Assets/Scripts/Scenarios/Fallback/Breadcrumbs/DisableBreadcrumbs.cs (hunk 1 above)
Comment (rich-bugsnag) [reply features/fixtures/maze_runner/Assets/Scripts/Scenarios/Fallback/Breadcrumbs/DisableBreadcrumbs.cs]: done 👍

**Review by nickdowell — APPROVED** [review]
LGTM!

File: Gemfile [hunk 3]
```diff
@@ -6,7 +6,7 @@ gem 'xcodeproj'
 
 unless Gem.win_platform?
   # Use official Maze Runner release
-  gem 'bugsnag-maze-runner', git: 'https://github.com/bugsnag/maze-runner', tag: 'v6.19.1'
+  gem 'bugsnag-maze-runner', git: 'https://github.com/bugsnag/maze-runner', tag: 'v7.2.1'
```
Comment (twometresteve) [inline Gemfile:9]: Upgrading to v7 at the same time is probably too much to take on in one PR - note that @Cawllec is already working on that.

**Pull request merged**